docs/documentation/development/bindings/discovery-services.md

```diff
@@ -0,0 +1,149 @@
+
+## Discovery 
+
+### Background Discovery 
+
+If the implemented discovery service enables background discovery, the `AbstractDiscoveryService` class automatically starts it. If background discovery is enabled, the framework calls `AbstractDiscoveryService#startBackgroundDiscovery` when the binding is activated and `AbstractDiscoveryService#stopBackgroundDiscovery` when the component is deactivated. The default implementations of both methods are empty and could be overridden by the binding developer. Depending on the concrete implementation the discovery service might start and stop a scheduler in these method or register a listener for an external protocol. The`thingDiscovered` method can be used to notify about a newly discovered thing.
+
+The following example shows the implementation of the above mentioned methods in the Wemo binding. 
+
+```java
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Start WeMo device background discovery");
+        if (wemoDiscoveryJob == null || wemoDiscoveryJob.isCancelled()) {
+            wemoDiscoveryJob = scheduler.scheduleAtFixedRate(wemoDiscoveryRunnable, 0, refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stop WeMo device background discovery");
+        if (wemoDiscoveryJob != null && !wemoDiscoveryJob.isCancelled()) {
+            wemoDiscoveryJob.cancel(true);
+            wemoDiscoveryJob = null;
+        }
+    }
+```
+
+### Active Scan
+
+If the user triggers an active scan for a binding or specific set of thing types, the method `startScan` of each discovery service which supports these thing type is called. Within these methods the things can be discovered. The abstract base class automatically starts a thread, so the implementation of this method can be long-running.
+
+The following example implementation for `startScan` is taken from the `HueLightDiscoveryService`, that triggers a scan for known and also for new lights of the hue bridge. 
+
+```java
+    @Override
+    public void startScan() {
+        List<FullLight> lights = hueBridgeHandler.getFullLights();
+        if (lights != null) {
+            for (FullLight l : lights) {
+                onLightAddedInternal(l);
+            }
+        }
+        // search for unpaired lights
+        hueBridgeHandler.startSearch();
+    }
+```
+
+### Remove older results
+
+Normally, older discovery results already in the inbox are left untouched by a newly triggered scan. If this behavior is not appropriate for the implemented discovery service, one can override the method `stopScan` to call `removeOlderResults` as shown in the following example from the Hue binding: 
+
+```java
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+```
+
+## UPnP Discovery
```

Comment:
It would have been nice to also have a section for the mDNSDiscoveryService here!